include/glow/Graph/Context.h

```diff
@@ -37,19 +38,43 @@ class Context final {
 public:
   /// Maps placeholders to the tensors that back them.
   using PlaceholderMap = std::unordered_map<Placeholder *, Tensor *>;
+  /// Additional Tensors used for this graph, and whether the memory is owned by
+  /// this Context.
+  using TensorMap = std::unordered_map<std::string, std::pair<Tensor *, bool>>;
 
 private:
   /// Maps Placeholders to Tensors.
   PlaceholderMap map_;
+  /// Maps node names to Tensors.
+  TensorMap tensorLookup_;
 
 public:
   /// \returns the tensor that corresponds to Placeholder \p P or Null if the
   /// tensor is not found.
-  Tensor *get(Placeholder *P) const;
+  Tensor *get(const Placeholder *P) const;
+
+  /// \returns the tensor that corresponds to the name \name, or Null if the
+  /// tensor is not found.
+  Tensor *get(std::string name) const;
 
   /// Inserts the Placeholder-Tensor pair.
   void insert(Placeholder *P, Tensor &&T);
 
+  /// Inserts the Placeholder-Tensor pair.
+  void insert(Placeholder *P, Tensor *T);
+
+  /// Attaches the tensor to the node \p N's name.
+  void registerTensor(const Named *N, Tensor *T, bool transferOwnership);
+
+  /// Attaches the tensor to the given name \p name.
+  void registerTensor(std::string name, Tensor *T, bool transferOwnership);
+
+  /// Detaches the Tensor of the given name and frees memory if owned.
+  void deleteTensor(const Named *N);
+
+  /// Detaches the Tensor of the given name and frees memory if owned.
+  void deleteTensor(std::string name);
```

Comment:
ditto

Reply:
ditto for which comment? the StringRef one or the ownership question?

Reply:
:P StringRef